kbase-extension/static/kbase/js/common/cellComponents/filePathWidget.js

```diff
@@ -663,6 +683,97 @@ define([
             });
         }
 
+        /**
+         * Steps.
+         * 1. Get all values from output parameters, ignore null values
+         * 2. Filter down to those with duplicates
+         * 3. Send those to just the inputs those came from.
+         * 4. Up to widgets to act accordingly
+         * 5. Return the row ids that are duplicates
+         */
+        function updateDuplicateOutputValues() {
+            const outputValueCounts = model.getItem('outputParamIds').reduce((counts, paramId) => {
+                const outputValues = getParameterValuesByRow(paramId);
+                for (const [rowId, value] of Object.entries(outputValues)) {
+                    if (!value) {
+                        return counts;
+                    } else if (!(value in counts)) {
+                        counts[value] = [];
+                    }
+                    counts[value].push({ rowId, paramId });
+                }
+                return counts;
+            }, {});
+            const values = Object.keys(outputValueCounts).reduce((duplicateValues, value) => {
+                if (outputValueCounts[value].length > 1) {
+                    duplicateValues[value] = outputValueCounts[value];
+                }
+                return duplicateValues;
+            }, {});
+            // values should now look like:
+            // {
+            //   'field_value': [{ rowId1, paramId }, { rowId2, paramId }]
+            // }
+            // with only entries where there are at least two row ids
+            // translate to something easier to work with -
+            // {
+            //    rowId: {
+            //      paramId1: {
+            //        widget: Object
+            //        duplicateRows: [rowId, rowId]
+            //      },
+            //      paramId2: {
+            //        widget: Object,
+            //        duplicateRows: [rowId, rowId]
+            //    }
+            // }
+            const duplicates = {};
+            Object.values(values).forEach((dupArr) => {
+                const allRows = dupArr.reduce((rowIds, entry) => {
+                    rowIds.add(dataModel.rowIdToIndex[entry.rowId] + 1);
+                    return rowIds;
+                }, new Set());
+                dupArr.forEach((entry) => {
+                    const rowId = entry.rowId;
```

Comment:
Each `entry`, as above, is a rowId (UUID) and parameter id.
This gets converted into:
```
{
    widget: the actual widget object,
    duplicateRows: Array of all the row numbers except for this one. (unless there's no rows, in which case we include this one. The case where there would be no rows would be if the duplicates are from another tab - but that's not supported yet, this just sets up that case).
}
```